crates/sui-json-rpc/src/error.rs

```diff
@@ -94,24 +100,103 @@ impl Error {
+                ErrorObject::owned(TRANSIENT_ERROR_CODE, err.to_string(), None::<()>);
+            RpcError::Call(CallError::Custom(error_object))
+        }
+        QuorumDriverError::ObjectsDoubleUsed {
+            conflicting_txes,
+            retried_tx,
+            retried_tx_success,
+        } => {
+            let error_message = format!(
+                "Failed to sign transaction by a quorum of validators because of locked objects. Retried a conflicting transaction {:?}, success: {:?}",
+                retried_tx,
+                retried_tx_success
+            );
+
+            let mut new_map: BTreeMap<TransactionDigest, Vec<ObjectRef>> = BTreeMap::new();
+
+            for (digest, (pairs, _)) in conflicting_txes {
+                let mut new_vec = Vec::new();
+
+                for (_authority, obj_ref) in pairs {
+                    new_vec.push(obj_ref);
+                }
+
+                new_map.insert(digest, new_vec);
+            }
+
+            let error_object =
+                ErrorObject::owned(INVALID_PARAMS_CODE, error_message, Some(new_map));
+            RpcError::Call(CallError::Custom(error_object))
+        }
+        QuorumDriverError::NonRecoverableTransactionError { errors } => {
+            let new_errors: Vec<String> = errors
+                .into_iter()
+                .filter(|(err, _, _)| !err.is_retryable().0) // consider retryable errors as transient errors
+                .map(|(err, _, _)| {
+                    match err {
+                        // Use inner UserInputError's Display instead of SuiError::UserInputError, which renders UserInputError in debug format
+                        SuiError::UserInputError { error } => error.to_string(),
+                        _ => err.to_string(),
+                    }
+                })
+                .collect();
+
+            let mut error_code = TRANSACTION_EXECUTION_CLIENT_ERROR_CODE;
+            let mut error_msg = "Transaction execution failed due to issues with transaction inputs. Please review the 'data' field and try again.";
+
+            if new_errors.is_empty() {
+                error_code = TRANSIENT_ERROR_CODE;
+                error_msg =
+                    "Transaction execution failed due to transient errors. Please try again."
+            }
+
+            let error_object = ErrorObject::owned(error_code, error_msg, Some(new_errors));
+            RpcError::Call(CallError::Custom(error_object))
+        }
+        QuorumDriverError::QuorumDriverInternalError(_) => {
+            let error_object = ErrorObject::owned(
+                INTERNAL_ERROR_CODE,
+                "Internal error occurred while executing transaction.",
+                None::<()>,
+            );
+            RpcError::Call(CallError::Custom(error_object))
+        }
+        QuorumDriverError::SystemOverload { .. } => {
+            let error_object = ErrorObject::owned(INTERNAL_ERROR_CODE, err.to_string(), None::<()>);
```

Comment:
Is this an internal error or a transient error?

Reply:
I assumed internal but I suppose it's similar to others like TooManyTransactionsPendingExecution etc?

Reply:
I would have thought that was a transient error as well? Both of these errors are temporary things that happen when the network is under too much load -- i.e. things are operating correctly and because of that we have detected too many transactions in flight and have stopped accepting more.

cc @longbowlu for a final verdict here as the expert.